include/gridtools/sid/loop.hpp

```diff
@@ -637,6 +638,30 @@ namespace gridtools {
             return {};
         }
 
+        template <class Key, int UnrollFactor, class NumSteps, class Step = integral_constant<int, 1>>
+        constexpr GT_FUNCTION auto make_unrolled_loop(NumSteps num_steps, Step step = {}) {
+            using u = integral_constant<int, UnrollFactor>;
```

Comment:
I am confused, why not use `UnrollFactor` directly? What's `u()`, isn't it just `u::value`, which is just `UnrollFactor`?

Reply:
No, `u()` is of type `integral_constant`. That is, if `num_steps` or `step` are also of type `integral_constant`, the arguments to `make_loop` on the following lines will be `integral_constant` and thus guaranteed compile-time evaluated, too.  If we directly take `UnrollFactor`, we don’t have this guarantee and fall into different overloads of `make_loop`.

Reply:
Ah makes sense!